docs/essentials/controls.md

```diff
@@ -0,0 +1,208 @@
+
+It does not require any modification to your components, and stories for controls are:
+
+- Convenient. Auto-generate controls based on React/Vue/Angular/etc. components.
+- Portable. Reuse your interactive stories in documentation, tests, and even in designs.
+- Rich. Customize the controls and interactive data to suit your exact needs.
+
+To use the controls addon, you need to write your stories using [args](../writing-stories/args). Storybook will automatically generate controls based on your args and what it can infer about your component; but you can configure the controls further using [argTypes](../api/stories#argtypes), see below.
+
+<div style="background-color:#F8FAFC">
+ask tom/dom/michael for the link of the migration guide.
+</div>
+
+> If you have written stories in the older pre-Storybook 6 style, you may want to read the [args&controls migration guide] to help understand how to convert your stories for args.
+
+### Choosing the control type
+
+By default, Storybook will choose a control for each arg based on the initial value of the arg. This works well with some kind of args, such as boolean values or free-text strings, but in other cases you want a more restricted control.
+
+For instance, suppose you have a `backgroundColor` arg on your story:
+
+```js
+// Button.stories.js
+
+const Red = ButtonStory.bind({});
+Red.args = {
+   backgroundColor: '#e00',
+};
+```
+
+By default, Storybook will render a free text input for the `backgroundColor` arg:
+
+<div style="background-color:#F8FAFC">
+TODO add same image used in the SB 6.0 docs with highlighted background control
+</div>
+
+This works as long as you type a valid string into the auto-generated text control, but it's not the best UI for picking a color. Let’s replace it with Storybook’s color picker component.
+
+We can specify which controls get used by declaring a custom [argType](../api/stories#argtypes) for the `backgroundColor` property. ArgTypes encode basic metadata for args, such as name, description, defaultValue for an arg. These get automatically filled in by Storybook Docs.
+
+ArgTypes can also contain arbitrary annotations which can be overridden by the user. Since `backgroundColor` is a property of the component, let's put that annotation on the default export.
+
+```js
+import { Button } from './Button';
+export default {
+  title: 'Button',
+  component: Button,
+  argTypes: {
+    backgroundColor: { control: 'color' },
+  },
+};
+```
+
+This replaces the input with a color picker leading to a more intuitive developer experience.
+
+<div style="background-color:#F8FAFC">
+TODO add same image used in the SB 6.0 docs color picker opened
+</div>
+
+### Fully custom args
+
+<div style="background-color:#F8FAFC">
+ask tom for location of complex stories link
```

Comment:
Link to the complex stories section of workflows.

Reply:
As the other item that is also pointing to this. I'm leaving the complex stories link to the workflows. In there should the reader want to know more about the item. It will redirect him to the writing stories.